### DISCUSSION TITLE
Show shape data in the diagram view

### DISCUSSION THREAD

**Discussion by ivan-pinatti** [jgraph/drawio#4380] [Q&A]
Hello,

I am trying to add _data_ to a shape and then have it always visible in the diagram view, I couldn't find this option anywhere. Is that possible somehow?

Example, I create a router and add several data items like; IP, mask, gateway, etc... And those are visible without having to hover over the shape.

I am trying to do this to have a more structured shape object type.

Thank you in advance.

**Comment by MJWootton** [comment]
If you want to show data on a shape, enable "placeholders" in the shape data, then add the appropriate tag in the shape text. For example, if you have a data item called `field`, write `%field%` on the shape and it will be replaced with the stored value when you exit editing the text.
